script/zip-comp-logs

```diff
@@ -4,27 +4,53 @@ A script to zip the logs produced by a series of competition matches.
 """
 
 import re
-import shutil
 import argparse
 import datetime
 from typing import List
 from pathlib import Path
 from zipfile import ZipFile, ZIP_DEFLATED
 
 
-def zip_folder(folder: Path, output_dir: Path, suffix: str, include_date: bool) -> str:
+def match_animation_files(log_name: str, archives_dir: Path) -> List[Path]:
+    match_num_search = re.search(r'match-([0-9]+)', log_name)
+    if not isinstance(match_num_search, re.Match):
+        print(f'Invalid match name: {log_name}')
+        return []
+    match_num = match_num_search[1]
+    match_files = (archives_dir / 'recordings').glob(f'match-{match_num}.*')
+    return [data_file for data_file in match_files if data_file.suffix != '.mp4']
+
+
+def zip_folder(
+    folder: Path,
+    output_dir: Path,
+    archives_dir: Path,
+    suffix: str,
+    include_date: bool,
+    animations: str,
+) -> str:
     tla = folder.name
     zip_name = f'team-{tla}{suffix}'
 
     if include_date:
         zip_name += f'-{datetime.date.today()}'
 
-    shutil.make_archive(
-        f'{output_dir / zip_name}',
-        'zip',
-        root_dir=folder,
-        base_dir='.',
-    )
+    with ZipFile(output_dir / f'{zip_name}.zip', 'w', compression=ZIP_DEFLATED) as zipfile:
+        for log in folder.resolve().iterdir():
+            zipfile.write(log, log.name)
+
+            if animations == 'team':
+                for animation_file in match_animation_files(log.name, archives_dir):
+                    zipfile.write(animation_file.resolve(), animation_file.name)
+
+        if animations == 'team':
+            # save animation textures
+            for texture in (archives_dir / 'recordings/textures').glob('**/*'):
+                zipfile.write(
+                    texture.resolve(),
+                    texture.relative_to(archives_dir / 'recordings/'),
```

Comment:
Does this need the trailing slash?

Reply:
No, I just tend to leave the trailing slash on directories